pkg/ruler/config.go

```diff
@@ -48,14 +48,25 @@ func (c *Config) Validate() error {
 }
 
 type RemoteWriteConfig struct {
-	Client              config.RemoteWriteConfig `yaml:"client"`
-	Enabled             bool                     `yaml:"enabled"`
-	ConfigRefreshPeriod time.Duration            `yaml:"config_refresh_period"`
+	Client              *config.RemoteWriteConfig  `yaml:"client,omitempty"`
+	Clients             []config.RemoteWriteConfig `yaml:"clients"`
+	Enabled             bool                       `yaml:"enabled"`
+	ConfigRefreshPeriod time.Duration              `yaml:"config_refresh_period"`
 }
 
 func (c *RemoteWriteConfig) Validate() error {
-	if c.Enabled && c.Client.URL == nil {
-		return errors.New("remote-write enabled but client URL is not configured")
+	if c.Enabled {
+		if (c.Client == nil || c.Client.URL == nil) && len(c.Clients) == 0 {
+			return errors.New("remote-write enabled but no clients URL are configured")
+		}
+
+		if len(c.Clients) > 0 {
+			for _, clt := range c.Clients {
+				if clt.URL == nil {
+					return fmt.Errorf("remote-write enabled but client '%s' URL is not configured", clt.Name)
+				}
+			}
+		}
 	}
```

Comment:
Suggest to flip the check to `!c.Enabled` and early return `nil`. This will save you one level of indentation and make the happy code path easier to read.